### PR TITLE
Unify conv

### PR DESCRIPTION
**Pull request by wangyang59**
issue #417 
Changed the usage of weights in ExpandConvBaseLayer to match the behaviour in CudnnConvLayer so that the model trained with both methods can be used interchangeably. 

Added a test_ConvUnify to verify the above desired property.

### CODE REVIEW

File: paddle/gserver/tests/test_ConvUnify.cpp [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                       /*channel*/ 1,
+                       /*numfilters*/ 2,
+                       input, param, true);
+    checkMatrixEqual(resultCpu, resultGpu);
+
+    input = Matrix::create(1, 3 * 3 * 2, false, false);
+    float inputData2[] = {1, 2, 3,
+                          4, 5, 6,
+                          7, 8, 9,
+
```
Comment (luotao1) [inline paddle/gserver/tests/test_ConvUnify.cpp]: 132行为什么要空一行

**Comment by wangyang59** [comment]
Hi @qingqing01 @luotao1 
I have added a unit-test in test_NetworkCompare per your request. 
As I was working on this issue, I found that there is another bug in cudnn_conv layer when groups>1. I fixed this bug too and add a corresponding unit-test in test_LayerGrad

File: paddle/gserver/tests/test_LayerGrad.cpp [hunk 2]
```diff
@@ -213,6 +213,53 @@ TEST(Projection, conv) {
       NUM_FILTERS,
       true);
 }
+
+TEST(Projection, conv2) {
```
Comment (luotao1) [inline paddle/gserver/tests/test_LayerGrad.cpp]: 单测里的conv2和之前的conv有什么区别么

File: paddle/gserver/tests/img_conv2_a.conf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from paddle.trainer_config_helpers import *
+
+settings(batch_size=10)
+data = data_layer(name ="input", size=8*16*16)
```
Comment (luotao1) [inline paddle/gserver/tests/img_conv2_a.conf]: img_conv2_a.conf和img_conv_a.conf非常像。能复用img_conv_a.conf，然后把img_conv2_b.conf改成img_conv_c.conf么

Comment (wangyang59) [inline paddle/gserver/tests/img_conv2_a.conf]: I felt that the purposes of the two tests are different so I would prefer to keep them separate.

File: paddle/gserver/tests/test_LayerGrad.cpp (hunk 2 above)
Comment (luotao1) [inline paddle/gserver/tests/test_LayerGrad.cpp]: 如果只有group有变化，这两个单测可以合并

File: paddle/gserver/tests/img_conv2_a.conf (hunk 3 above)
Comment (luotao1) [inline paddle/gserver/tests/img_conv2_a.conf]: 目的是不一样的，所以可以在test_NetworkCompare.cpp调用时说一下。配置不用写两份。

**Comment by luotao1** [comment]
如果这个bug很着急修复的话，可以先merge，然后在下一个PR中完善下单测。@qingqing01 @wangyang59

**Comment by wangyang59** [comment]
Hi @luotao1 
I have refactored the two unit-tests in test_LayerGrad and consolidated the img_conv.conf in test_NetworkCompare following your requests.

**Pull request merged**